### PR TITLE
Migrate almost all remaining code to ES6 classes

### LINKED ISSUE #2692: Leftover class to be converted to ES6
Files that still need to be converted:

- [x] asset-reference
- [x] asset-variants
- [ ] channel
- [ ] channel3d
- [x] object-pool - rename to allocate-pool?
- [x] stats
- [x] program-library
- [x] vertex-format
- [x] I18nParser
- [x] controller
- [x] element-input
- [x] game-pads
- [x] glb-parser
- [x] audio
- [x] untar
- [x] script-attributes
- [x] script-type
- [ ] instance
- [ ] instance3d
- [x] sound
- [x] xr-dom-overlay
- [x] xr-finger
- [x] xr-joint

### PR DESCRIPTION
**Pull request by willeastcott**
The only modules this PR doesn't address are audio/sound related ones and `vertex-format.js`. These probably require some level of refactoring to elegantly migrate.

Fixes #2692

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/graphics/vertex-format.js [hunk 1]
```diff
@@ -105,117 +105,111 @@ import {
...
- * @description Returns {@link pc.VertexFormat} used to store matrices of type {@link pc.Mat4} for hardware instancing.
- */
-Object.defineProperty(VertexFormat, 'defaultInstancingFormat', {
-    get: (function () {
-        return function () {
-            return this._defaultInstancingFormat;
-        };
-    }())
-});
+        this._defaultInstancingFormat = new VertexFormat(graphicsDevice, formatDesc);
```
Comment (mvaligursky) [inline src/graphics/vertex-format.js:187]: I'd expect _defaultInstancingFormat to be static as well .. not sure how come it wasn't before and how I got this to work. I'm surprised that static init function call access this? I'll have to play with this, but feels wrong (starting with my original code).

Comment (willeastcott) [reply src/graphics/vertex-format.js]: I've revised the code.

Comment (mvaligursky) [reply src/graphics/vertex-format.js]: That looks great now. I have no idea how my original code worked.

File: src/asset/asset.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ const VARIANT_DEFAULT_PRIORITY = ['pvr', 'dxt', 'etc2', 'etc1', 'basis'];
  * @property {pc.AssetRegistry} registry The asset registry that this Asset belongs to
  */
 class Asset extends EventHandler {
-    constructor(name, type, file, data = {}, options = {}) {
```
Comment (Maksims) [inline src/asset/asset.js:74]: Why default properties not preferred here?

Comment (willeastcott) [reply src/asset/asset.js:74]: It's breaks texture loading and I have no idea why yet, so I backed it out for now to get this PR merged.

File: src/graphics/vertex-format.js (hunk 1 above)
Comment (willeastcott) [reply src/graphics/vertex-format.js]: Cool. Happy for me to merge then?

**Review by mvaligursky — APPROVED** [review]
looks great to me.

**Pull request merged**